src/test/java/com/school/project/controller/LessonControllerTest.java

```diff
@@ -0,0 +1,247 @@
+                .andDo(print())
+                .andExpect(jsonPath("$.[0].id").value(12))
+                .andExpect(jsonPath("$.[0].subjectId").value(3))
+                .andExpect(jsonPath("$.[0].date").value("2019-05-04T00:00:00.000+0000"))
+                .andExpect(jsonPath("$.[0].groupId").value(3))
+                .andExpect(jsonPath("$.[0].teacherId").value(2))
+                .andExpect(jsonPath("$.[1].id").value(11))
+                .andExpect(jsonPath("$.[1].subjectId").value(3))
+                .andExpect(jsonPath("$.[1].date").value("2019-05-04T00:00:00.000+0000"))
+                .andExpect(jsonPath("$.[1].groupId").value(3))
+                .andExpect(jsonPath("$.[1].teacherId").value(2));
+    }
+
+    @Test
+    public void testCreateNewLesson() throws Exception {
+        when(lessonService.createLesson(getRealTestLesson())).thenReturn(getTestLesson());
+        LessonDto lessonDto = getTestLessonDto();
+        mvc.perform(post("/lessons", lessonDto)
+                .content(NEW_LESSON_JSON_STRING2)
+                .contentType(MediaType.APPLICATION_JSON_UTF8))
+                .andExpect(status().isOk());
+
+//        DateFormat format = new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss");
+//        Date date = format.parse("2019-05-04T");
+        verify(lessonService).createLesson(new Lesson(null, "Angular",
+                new Subject(3L, null), null,
+                new Group(3L, null, null, null),
+                new User(3L, null, null, null, null, null, null, null)));
+    }
+
+    @Test
+    public void testUpdateLesson() throws Exception {
+        when(lessonService.updateLesson(getLessonToUpdate(), 12L)).thenReturn(getTestLesson());
+        mvc.perform(put("/lessons/" + 12L).content(NEW_LESSON_JSON_STRING2)
+                .contentType(APPLICATION_JSON))
+                .andExpect(status().isOk())
+                .andDo(print())
+                .andExpect(jsonPath("$.id").value(12))
+                .andExpect(jsonPath("$.subjectId").value(3))
+                .andExpect(jsonPath("$.date").value("2019-05-04T00:00:00.000+0000"))
+                .andExpect(jsonPath("$.groupId").value(3))
+                .andExpect(jsonPath("$.teacherId").value(2));
+        verify(lessonService).updateLesson(getLessonToUpdate(), 12L);
+    }
+
+    private Lesson getTestLesson() throws Exception {
+        DateFormat format = new SimpleDateFormat("yyyy-MM-dd");
+        date = format.parse("2019-05-04");
+
+        return new Lesson(12L,
+                "JS",
+                new Subject(3l, "QA"),
+                date,
+                new Group(3L, new Date(), new Module(), new ArrayList<User>()),
+                new User(2L, null, null, new Date(), null, null, new Date(), new Date()));
+    }
+
+    private Lesson getRealTestLesson() throws Exception {
+            DateFormat format = new SimpleDateFormat("yyyy-MM-dd");
+            date = format.parse("2019-05-04");
+
+            return new Lesson(null,
```

Comment:
The data (lesson) you pass to your lessonService.createLesson() method should be as the data you specified in your mock. 

In your case it was different, for instance, you mocked lesson with thema "JS" while you were passing  there lessone with the thema "Angular". And the same was for almost all the fields: for the Group you mocked non-null Date, Module and the list of users while you passed null-values for this. Compare your getTestLesson() method and my getRealTestLesson(). I think you may easy fix the rest of the tests, just make sure that what you mock and what you pass when invoking the test - the same data.